main.cpp

```diff
@@ -38,8 +39,275 @@
 #include <modules/embit/module.h>
 #endif
 
+#ifdef CUSTOM_MUTATOR_BOLT11
+#include "modules/bitcoin/bech32.h"
+#endif
+
 std::shared_ptr<bitcoinfuzz::Driver> driver = nullptr;
 
+#ifdef CUSTOM_MUTATOR_BOLT11
+extern "C" size_t LLVMFuzzerMutate(uint8_t *data, size_t size, size_t max_size);
+extern "C" size_t LLVMFuzzerCustomMutator(uint8_t *fuzz_data, size_t size, size_t max_size,
+			       unsigned int seed);
+extern "C" size_t LLVMFuzzerCustomCrossOver(const uint8_t *in1, size_t in1_size, const uint8_t *in2,
+				 size_t in2_size, uint8_t *out, size_t max_out_size,
+				 unsigned seed);
+
+// Encodes a dummy bolt11 invoice into `fuzz_data` and returns the size of the
+// encoded string.
+static size_t initial_input(uint8_t *fuzz_data, size_t size, size_t max_size)
+{
+    const std::string dummy =
+        "lnbc16lta047pp5h6lta047h6lta047h6lta047h6lta047h6lta047h6lta047"
+	    "h6lqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq"
+	    "qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqxnht6w";
```

Comment:
Nit: this can be a compile-time constant:

```suggestion
    constexpr std::string_view dummy =
        "lnbc16lta047pp5h6lta047h6lta047h6lta047h6lta047h6lta047h6lta047"
	    "h6lqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq"
	    "qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqxnht6w";
```

Reply:
Done